src/main/java/com/monumental/services/UserService.java

```diff
@@ -0,0 +1,38 @@
+package com.monumental.services;
+
+import com.monumental.controllers.helpers.CreateUserRequest;
+import com.monumental.exceptions.InvalidEmailOrPasswordException;
+import com.monumental.models.Role;
+import com.monumental.models.User;
+import com.monumental.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService extends ModelService<User> {
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public User signup(CreateUserRequest userRequest) throws InvalidEmailOrPasswordException {
+        if (this.userRepository.getByEmail(userRequest.getEmail()) != null) {
+            throw new InvalidEmailOrPasswordException("Email address already in use.");
+        }
+        if (!userRequest.getPassword().equals(userRequest.getMatchingPassword())) {
+            throw new InvalidEmailOrPasswordException("Passwords must match.");
+        }
+        User user = new User();
+        user.setEmail(userRequest.getEmail());
+        user.setPassword(passwordEncoder.encode(userRequest.getPassword()));
+        user.setRole(Role.COLLABORATOR);
+        user.setName(userRequest.getName());
+        // TODO: Disable once email verification is in place
```

Comment:
What do you mean by this?

Reply:
When we have email confirmation set up, user accounts should be disabled/not verified until they click the link in the email